### ISSUE TITLE
Device cheatsheet images

### ISSUE THREAD

**Issue by WhiteMagic** [WhiteMagic/JoystickGremlin#81]
With the addition of a library that allows generating arbitrary PDF files I will be able to render text in arbitrary locations on an image, which will allow me to create cheat sheets that place the text on drawings of devices.

As there are quite a few devices out there and many different drawings for such cheat sheets I'd be more then happy for people to link such images in here as a sort of repository for me to use when I get around to implementing that feature.

**Comment by Carlton2001** [comment]
Thank you for your hard work WhiteMagic, your software is really improved each update.

Below the PDF manual of the VPC MongoosT-50 Grip.
https://virpil.com/en/support/downloads?download=16:vpc-mongoost-50-grip-user-manual-en
Page 6 Left Grip, Page 7 Right Grip

![](https://virpil.com/images/blog/announcements/h8zX2kqMcGI.jpg)

**Comment by Wolfereign** [comment]
![t16000m](https://user-images.githubusercontent.com/5723368/35827203-5072a288-0a78-11e8-8194-4f92f4b068c1.png)
Put a "Left" or "Right" syntax at the bottom right and you can cover both configurations for the stick.  Just a disclaimer but this template is not my work but something I have found and use from the internet.

**Comment by ddrake1984** [comment]
I am really looking forward to this feature

**Comment by MikeSpaceG** [comment]
I had started on something like this for my own use when I came across this issue. Currently I piggy-backing off generate_cheatsheet() and call my own simple reference card builder function.

I will help implement a generic solution if you are accepting contributions.

**Comment by WhiteMagic** [comment]
Absolutely, I'm more then happy to add contributions from others into Gremlin. The current generator is really just a bare bones implementation that gets some information out but obviously has tons of directions into which it could be improved.

**Comment by WhiteMagic** [comment]
https://cdn.discordapp.com/attachments/438688364359581710/518876091373453315/MT-50_GripLH_ButtonMapR.png

**Comment by Psebcool** [comment]
Excellent idea. Can the user import their own layouts in the software or will they have to choose from a predefined list ?

**Comment by WhiteMagic** [comment]
There hasn't been any real work on this. However, someone else has written a program that implements this idea and his software is able to read Gremlin profiles and generate diagrams like this issue here is talking about. https://github.com/Rexeh/joystick-diagrams

**Comment by CircuitGuy** [comment]
FYI for any aspiring devs... https://github.com/Rexeh/joystick-diagrams is closing down now.

It might be worth pulling in a snapshot and bundling it with Gremlin. Obviously don't want the extra work on WhiteMagic right now who's in the middle of the UI refactor.
